assets/scripts/app/__tests__/flag_overrides.test.js

```diff
@@ -0,0 +1,81 @@
+// Test cases:
+// 1) User flags override default flags
+// 2) Session flags override user flags
+/* eslint-env jest */
+
+import { generateFlagOverrides, applyFlagOverrides } from '../flag_utils'
+
+jest.mock('../../app/routing', () => {})
+jest.mock('../../app/errors', () => {})
+
+const initialFlags = {
+  FOO_BAR: {
+    value: true,
+    source: 'initial'
+  },
+  BAZ_QUX: {
+    value: false,
+    source: 'initial'
+  }
+}
+
+const flagOverrides = [
+  {
+    source: 'user',
+    flags: [
+      {
+        flag: 'FOO_BAR',
+        value: false
+      }
+    ],
+    priority: 1
+  },
+  {
+    source: 'session',
+    flags: [
+      {
+        flag: 'BAZ_QUX',
+        value: true
+      },
+      {
+        flag: 'FOO_BAR',
+        value: true
+      }
+    ],
+    priority: 2
+  }
+]
+
+describe('generateFlagOverrides', () => {
+  it('creates an object with source, flags, and priority as keys', () => {
+    const userFlags = {
+      FOO_BAR: false
+    }
+
+    const result = generateFlagOverrides(userFlags, 'user')
+    expect(result).toEqual({
+      source: 'user',
+      flags: [{
+        flag: 'FOO_BAR',
+        value: false
+      }],
+      priority: 1
+    })
+  })
+})
+
+describe('applyFlagOverrides', () => {
+  it('updates feature flag values according to flag overrides and priority levels', () => {
+    const result = applyFlagOverrides(flagOverrides, initialFlags)
+    expect(result).toEqual({
+      FOO_BAR: {
+        source: 'session',
+        value: true
+      },
+      BAZ_QUX: {
+        source: 'session',
+        value: true
+      }
+    })
+  })
+})
```

Comment:
Let's add one more situation in the test.  Put a flag in `initialFlags` that has a value that isn't in `flagOverrides`, so it should pass through to the result unchanged. This is because the test, as currently written, could pass just by throwing away all override objects except for the one with the highest priority. We want to make sure the test passes with a combined result.

Reply:
Another situation to test would be to also add a flag in the "user" override object that isn't overridden in "session" flags.